bindings/rust/wasmedge-sys/src/raw_result.rs

```diff
@@ -0,0 +1,36 @@
+use super::wasmedge;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ErrReport {
+    pub code: u32,
+    pub message: &'static str,
+}
+
+pub fn is_ok(res: wasmedge::WasmEdge_Result) -> bool {
+    unsafe { wasmedge::WasmEdge_ResultOK(res) }
+}
+
+pub fn get_code(res: wasmedge::WasmEdge_Result) -> u32 {
+    unsafe { wasmedge::WasmEdge_ResultGetCode(res) }
+}
+
+pub fn get_message<'a>(res: wasmedge::WasmEdge_Result) -> &'a str {
+    unsafe {
+        std::ffi::CStr::from_ptr(wasmedge::WasmEdge_ResultGetMessage(res))
+            .to_str()
+            .unwrap_or("Utf8 Error")
+    }
+}
+
+// Since WasmEdge_ErrCode is subject to change on the wasmedge side
+// it does not correspond to enum here
+pub fn decode_result(res: wasmedge::WasmEdge_Result) -> Result<(), ErrReport> {
+    if is_ok(res) {
+        Ok(())
+    } else {
+        Err(ErrReport {
+            code: get_code(res),
+            message: get_message(res),
+        })
```

Comment:
May we use `impl From<wasmedge::WasmEdge_Result> for ErrReport` here?
Because the methods `get_code` and `get_message` are used the same time, it maybe be better use `From` trait without missing one of the method.

Reply:
good idea.